internal/iam/resource.go

```diff
@@ -49,28 +48,29 @@ type ResourceWithScope interface {
 
 // LookupScope looks up the resource's  scope
 func LookupScope(ctx context.Context, reader db.Reader, resource ResourceWithScope) (*Scope, error) {
+	const op = "iam.LookupScope"
 	if reader == nil {
-		return nil, stderrors.New("error reader is nil for LookupScope")
+		return nil, errors.New(errors.InvalidParameter, op, "nil reader")
 	}
 	if resource == nil {
-		return nil, stderrors.New("error resource is nil for LookupScope")
+		return nil, errors.New(errors.InvalidParameter, op, "missing resource")
 	}
 	if resource.GetPublicId() == "" {
-		return nil, fmt.Errorf("LookupScope: scope id is unset %w", errors.ErrInvalidParameter)
+		return nil, errors.New(errors.InvalidParameter, op, "missing public id")
 	}
 	if resource.GetScopeId() == "" {
 		// try to retrieve it from db with it's scope id
 		if err := reader.LookupById(ctx, resource); err != nil {
-			return nil, fmt.Errorf("unable to get resource by public id: %w", err)
+			return nil, errors.Wrap(err, op)
```

Comment:
hum.. not sure what this error will look like... what's the error string?

Reply:
This would become something like `iam.LookupScope: db.LookupById: error we got in lookup`, if I left the context in it would be: `iam.LookupScope: unable to get resource by public id: db.LookupById: error we got in lookup`  I figured here the `unable to get resource by public id` was equivalent to db.LookupById`